src/features/bivariate_manager/components/BivariateMatrixControl/components/QuotientSelector/QuotientSelector.tsx

```diff
@@ -132,17 +68,90 @@ export const QuotientSelector = memo(
     selectedQuotient,
     onSelectQuotient,
     type,
+    children,
   }: QuotientSelectorProps) => {
-    return quotients.length > 1 ? (
-      <QuotientSelectorEnabled
-        id={id}
-        quotients={quotients}
-        selectedQuotient={selectedQuotient}
-        onSelectQuotient={onSelectQuotient}
-        type={type}
+    const [headingState, setHeadingState] = useHeadingGlobalState();
+
+    const toggleVisibility = (e) => {
+      e.stopPropagation();
+      if (headingState.headingId === id) {
+        setHeadingState({ headingId: '', width: 0 });
+      } else {
+        const selectorPosition = e.currentTarget.getBoundingClientRect();
+        const wholeRowPosition =
+          e.currentTarget.parentElement.parentElement.getBoundingClientRect();
+        const width = Math.ceil(
+          (selectorPosition.left - wholeRowPosition.left) * SCALING_BACK + MODAL_MARGIN,
+        );
+        setHeadingState({ headingId: id, width });
+      }
+    };
+
+    const selectQuotient = useCallback(
+      (numId: string, denId: string) => {
+        setHeadingState({ headingId: '' });
+        onSelectQuotient(numId, denId);
+      },
+      [setHeadingState, onSelectQuotient],
+    );
+
+    const [selected, others] = useMemo(() => {
+      const selectedIndex = quotients.findIndex(
+        ({ id }) => id[0] === selectedQuotient.id[0] && id[1] === selectedQuotient.id[1],
+      );
+      if (selectedIndex >= 0) {
+        return [
+          quotients[selectedIndex],
+          quotients.filter((_, i) => i !== selectedIndex),
+        ];
+      }
+      return [null, quotients];
+    }, [quotients, selectedQuotient.id]);
+
+    const renderQuotientItem = (
+      { id, label: numeratorLabel, quality }: Quotient,
+      isSelected: boolean,
+    ) => (
+      <QuotientItem
+        key={JSON.stringify(id)}
+        onSelectQuotient={selectQuotient}
+        quality={quality}
+        numeratorId={id[0]}
+        denominatorId={id[1]}
+        numeratorLabel={numeratorLabel}
+        isSelected={isSelected}
       />
-    ) : (
-      <QuotientSelectorDisabled type={type} />
+    );
+
+    return (
+      <div
+        className={cn({
+          [styles.denominators]: true,
+          [styles.row]: type === 'horizontal',
+          [styles.column]: type === 'vertical',
+        })}
+      >
+        <div onClick={toggleVisibility}>{children}</div>
+        {headingState.headingId === id && (
+          <div
+            className={styles.denominatorsContainer}
+            style={
+              type === 'horizontal'
+                ? {
+                    left: -headingState.width,
+                    right: 0,
+                  }
+                : {
+                    top: -headingState.width,
+                    bottom: 0,
+                  }
+            }
+          >
+            {selected && renderQuotientItem(selected, true)}
+            {others.map((item) => renderQuotientItem(item, false))}
```

Comment:
i'd suggest rename "others" to smth more informative, for exapmle: otherOptions, options, rest, notSelected